librz/core/cmd/cmd_flag.c

```diff
@@ -299,115 +280,205 @@ static void spaces_list(RzSpaces *sp, RzOutputMode mode) {
 	}
 }
 
+static int cmpflag(const void *_a, const void *_b) {
+	const RzFlagItem *flag1 = _a, *flag2 = _b;
+	return (flag1->offset - flag2->offset);
+}
+
+static void flag_describe_delta(RzCore *core, RzFlagItem *f, ut64 addr, RzOutputMode mode) {
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		PJ *pj = pj_new();
+		pj_o(pj);
+		pj_kn(pj, "offset", f->offset);
+		pj_ks(pj, "name", f->name);
+		// Print flag's real name if defined
+		if (f->realname) {
+			pj_ks(pj, "realname", f->realname);
+		}
+		pj_end(pj);
+		rz_cons_print(pj_string(pj));
+		if (pj) {
+			pj_free(pj);
+		}
+	} else {
+		// Print realname if exists and asm.flags.real is enabled
+		if (core->flags->realnames && f->realname) {
+			rz_cons_printf("%s + %d\n", f->realname,
+				(int)(addr - f->offset));
+		} else {
+			rz_cons_printf("%s + %d\n", f->name,
+				(int)(addr - f->offset));
+		}
```

Comment:
As I see it:
```suggestion
		char *name = core->flags->realnames && f->realname ? f->realname : f->name;
		if (f->offset != addr) {
			rz_cons_printf("%s + %d\n", name, (int)(addr - f->offset));
		} else {
			rz_cons_println(name);
		}
```
No need in `flag_describe_precise()` and `flag_describe_delta()`.